### PR TITLE
Use API_PORT and --port option instead of PORT_SHIFT

### LINKED ISSUE #1642: Minor change to how API_PORT and API_URL are set
### Problem

To set a different port we are currently using an envvar called `PORT_SHIFT` that works like so:

```js
const applyPortShift = (env) => {
  // TODO: implement automatic port selection when `PORT_SHIFT` is 'auto'.
  const API_PORT = 8000 + Number.parseInt(env.PORT_SHIFT || '0')
  const API_URL = 'http://127.0.0.1:' + API_PORT

  if (Number.isNaN(API_PORT) || API_PORT < 8000 || API_PORT > 65535) {
    throw new RangeError(`Invalid API_PORT value: ${API_PORT}.`)
  }
  return { ...env, API_PORT, API_URL }
}

Object.assign(process.env, applyPortShift(process.env))
```

I'm not sure why I chose to go with `PORT_SHIFT` instead of simply specifying the port via `API_PORT` envvar.. we should be able to specify the port directly. However, the above code makes that impossible because `API_PORT` is always overwritten even if it's set.

### Solution

I think the solution is as simple as the following modification:

```js
  return { API_PORT, API_URL, ...env }
```

### PR DESCRIPTION
**Pull request by snowteamer**
Closes #1642

### Summary of changes:
- New `API_PORT` envar, allowing to specify the default backend port. If not defined, then port 8000 will be the default.
- New `--port` Grunt option, to be used in Grunt commands. Useful e.g. when running two or more server instances.
 Example: `grunt --port=8004` will try to launch a dev instance on port 8004, whereas the usual `grunt` command will rely on the `API_PORT` envar, or 8000 if not defined.
- Cypress tests have been updated to no longer refer to a hard-coded `http://localhost:8000 URL`.
  This allows to run them against a backend running on another port.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -66,6 +45,10 @@ const {
   EXPOSE_SBP = ''
 } = process.env
 
+if (!['development', 'production'].includes(NODE_ENV)) {
+  throw new TypeError(`Invalid NODE_ENV value: ${NODE_ENV}.`)
+}
```
Comment (Silver-IT) [inline Gruntfile.js:47]: It's a nice checking for exceptions.

File: frontend/views/pages/DesignSystem.vue [hunk 2]
```diff
@@ -1306,7 +1306,7 @@ page(
             pre
               | link-to-copy.c-link(link='string')
           td
-            link-to-copy.c-link(link='http://localhost:3000/app/join?groupId=21XWnNJvq9&secret=3763')
+            link-to-copy.c-link(link='/app/join?groupId=21XWnNJvq9&secret=3763')
```
Comment (Silver-IT) [inline frontend/views/pages/DesignSystem.vue:1309]: Nice fix.

**Review by Silver-IT — DISMISSED** [review]
Great work, @snowteamer. I totally approve this PR. :)

**Review by taoeffect — APPROVED** [review]
LGTM! Just needs merge conflict fix.

**Review by taoeffect — CHANGES_REQUESTED** [review]
Marking as "changes requested" until merge conflict is fixed so that it correctly shows up in the list of PRs

File: test/cypress/integration/group-paying.spec.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ export function humanDate (datems, opts = { month: 'short', day: 'numeric' }) {
   return new Date(datems).toLocaleDateString(locale, opts)
 }
 
+const API_URL = Cypress.config('baseUrl')
```
Comment (taoeffect) [inline test/cypress/integration/group-paying.spec.js:8]: Nice 👍

**Review by taoeffect — APPROVED** [review]
Approved! Great work @snowteamer!!

**Pull request merged**